### DISCUSSION TITLE
Formprovider as HOC

### DISCUSSION THREAD

**Discussion by dancixx** [react-hook-form/react-hook-form#5740] [Q&A]
Hi guys,

I am new in react-hook-form, and my question, that when I wrap my whole application with FormProvider can I access the state like the react context API, or the redux provider or this component is just for inject the useform methods into the context for calling everywhere?

My case is that I have a react-native application. I wrapped my app into the FormProvider in the App.tsx. I have a form screen where I have to add some data from another screen (react-native-screens). So I tried to call getValues and setValue on the new screen but I get undefined as a result of the getValues method.
Should I create custom react context with useEffect to update the form state or there is chance to solve this with useFormContext hook?

**Comment by bluebill1049** [comment]
I think you want a state management library to host those submitted values. useFormContext only give context for those hook form methods.
